src/api/spec/controllers/webui/project_controller_spec.rb

```diff
@@ -687,4 +691,34 @@
       is_expected.to redirect_to(project_show_path(user.home_project))
     end
   end
+
+  describe 'POST #new_incident_request' do
+    before do
+      login user
+      request.env["HTTP_REFERER"] = root_url # Needed for the redirect_to :back
+    end
+
+    it "without an existent project will raise an exception" do
+      expect { post :new_incident_request, project: 'non:existent:project' }.to raise_error Project::UnknownObjectError
+    end
+
+    context "without a proper action for the maintenance project" do
+      before do
+        post :new_incident_request, project: maintenance_project, description: "Fake description for a request"
+      end
+
+      it { expect(flash[:error]).to eq("MaintenanceHelper::MissingAction") }
+      it { is_expected.to redirect_to(:back) }
+    end
+
+    context "with the proper params" do
+      before do
+        BsRequest.any_instance.stubs(:save!).returns(true)
+        post :new_incident_request, project: maintenance_project, description: "Fake description for a request"
```

Comment:
@mdeniz @ChrisBr One thing I realized is that you use stub here, not even expects... So there is nothing that garanties the object being saved.
So there is nothing that would ensure that BsRequest get's created and that it has the correct data.
Right now one can change the action to this and it would pass the test:
```
  def new_incident_request
    flash[:success] = 'Created maintenance incident request'
    redirect_to :action => 'show', :project => params[:project]
  end
```